src/main/java/org/terasology/blockNetwork/EfficientBlockNetwork.java

```diff
@@ -35,34 +40,82 @@
 
     private boolean mutating;
 
+    /**
+     * Adds a listener to topology events.
```

Comment:
"Adds *the given* listener"..

How do you add a listener *to* events? This makes no sense. Have a look at this page: https://docs.oracle.com/javase/8/javafx/api/javafx/beans/value/ObservableValue.html#addListener-javafx.beans.value.ChangeListener-

Clearly specify when the listener will be notified, and what happens when registering the same listener twice. (in this case, since `listeners` is a HashMap and you're not checking for existence before calling `add()`, the method will fail silently, meaning adding a duplicate listener has no functional effect.